### PR TITLE
Feature/task 3

### PR DESCRIPTION
**Pull request by kubmir**

### CODE REVIEW

File: test/reducers/notes/list/addListMember.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { isAddingNote } from '../../../../src/reducers/notes/list/isAddingNote';
```
Comment (vit-svoboda) [inline test/reducers/notes/list/addListMember.test.js]: The test files should be named after the thing they test, in this case, `isAddingNote.test.js`

File: test/reducers/notes/list/addListMember.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import {
+  prepareAddListMemberInitialState,
+} from '../../../testUtils/prepareTestData';
```
Comment (vit-svoboda) [inline test/reducers/notes/list/addListMember.test.js]: I'd declare this in the test, there's no reuse value. Zuzka even advocates that the initial state of a reducer should be declared only in the default value of the state parameter of the reducer.

File: test/reducers/notes/list/addListMember.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  let initialState;
+
+  beforeEach(() => {
+    initialState = prepareAddListMemberInitialState();
+  });
+
+  it('should return previous state if unknown action is dispatched', () => {
+    const unknownAction = {
+      type: 'Test action',
+      payload: {},
```
Comment (vit-svoboda) [inline test/reducers/notes/list/addListMember.test.js]: The action doesn't have to carry a payload. So this line is unnecessary.

File: test/selectors/notes/list/listOfNotes.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    expect(actualNote).toEqual(expectedNote);
+  });
+
+  it('getAllIds test', () => {
+    const notes = prepareNotesInitialState();
+    const expectedIds = [1, 2];
+
+    const actualIds = getAllIds(notes);
+
+    expect(actualIds).toEqual(expectedIds);
```
Comment (vit-svoboda) [inline test/selectors/notes/list/listOfNotes.test.js]: While this test is fine, you should also test the memoization - that the function returns the same object if you call it multiple times.

File: src/components/List.jsx [hunk 5]
```diff
@@ -1,120 +1,31 @@
...
-              <AddListMember
-                onAddClick={this.addNewNote}
-                onInputFocus={this.onIsAddListMemberFocus}
-                onInputBlur={this.onIsAddListMemberBlur}
-                isInputFocused={this.state.isAddListMemberFocused}
-              />
-            </li>
-          </ul>
-        </div>
+      <div>
```
Comment (vit-svoboda) [inline src/components/List.jsx]: You render this `List` in a `<ul>` element, that should only contain `<li>` elements - [don't take my word for it](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul). If you threw your output markup at a validator, I believe you'd get an error.

The solution here is either upgrading React to version 16 so you can [return fragments from a render](https://reactjs.org/blog/2017/09/26/react-v16.0.html#new-render-return-types-fragments-and-strings). Or you can restructure your components so the whole `<ul>` with all `<li>` are together in a single component.

**Comment by kubmir** [comment]
Now, I store an id of note to the database as noteId, Note object contains id property and other ids in the application are named noteId. 

I agree with possible renaming of ListMembers to Notes. I will do it in task 4 if it is possible because now it would be harder to rename files in task 3 and also in task 4 branch - I would do the renaming twice.

**Pull request merged**